scripts/utils.sh

```diff
@@ -17,7 +17,7 @@ data_dir="${data_home}/ubports-installer-cli"
 
 fetch() {
   url=${1}
-  cached="${cache_dir}/${url}"
+  cached="${cache_dir}/$(basename "${url}")"
```

Comment:
This will break horribly, there is no guarantee the filenames that will be fetched are unique. The cache directory is not supposed to be pretty, having full URLs there doesn't hurt anything.

Reply:
> This will break horribly, there is no guarantee the filenames that will be fetched are unique. The cache directory is not supposed to be pretty, having full URLs there doesn't hurt anything.

The old one actually breaks worse, for example when it downloads the channels.json from https://system-image.ubports.com, the command manages to separate the link, 'https:' becomes a directory in which '/system-image.ubports.com' exists and then finally comes the channels.json.
And it won't refetch the file anyway, so I don't see how it breaks...?

Reply:
That's not breaking, that's the way it's supposed to work and there is nothing wrong with that - like I said, the directory structure is not supposed to be pretty. And what I mean is that you can for example download the channels list for one device, but since you used `basename` if you try to do it for another it will use the cached one and you will download files for a wrong device.

Reply:
ah, I see, yeah I see how it breaks now